lib/BackbeatMetadataProxy.js

```diff
@@ -105,6 +105,58 @@ class BackbeatMetadataProxy extends BackbeatTask {
         });
     }
 
+    headLocation(params, log, cb) {
+        this.retry({
+            actionDesc: 'head object location from source',
+            logFields: {
+                bucket: params.bucket,
+                objectKey: params.objectKey
+            },
+            actionFunc: done => this._headLocationOnce(params, log, done),
+            shouldRetryFunc: err => err.retryable,
+            log,
+        }, cb);
+    }
+
+    _headLocationOnce(params, log, cb) {
+        const { bucket, objectKey, locations } = params;
+        log.debug('heading object location', {
+            where: 'source',
+            bucket, objectKey, locations,
+            method: 'BackbeatMetadataProxy._headLocationOnce',
+        });
+        const cbOnce = jsutil.once(cb);
+        const req = this.backbeatSource.multipleBackendHeadObject({
```

Comment:
Ideally we would have called this route "multipleBackendHeadLocation" as well, we could fit this in a later refactor.